File: 2015/j2/main.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
from pathlib import Path

def read_input(file_name: str):
    """Lit un fichier situé dans le même dossier que le script appelant"""
    path = Path(__file__).resolve().parent / file_name
    with path.open("r", encoding="utf-8") as f:
        return [line.rstrip("\n") for line in f]

def part1():
    input_name = read_input("input.txt")

    foot = 0

    for i in input_name:
        values = list(map(int, i.split("x")))
        values.sort()

        additionnal = int(values[0])*int(values[1])
        surfaces = 2*(int(values[0])*int(values[1]) + int(values[0])*int(values[2]) + int(values[1])*int(values[2]))

        foot += additionnal + surfaces

    return foot

def part2():
    input_name = read_input("input.txt")

    foot = 0

    for i in input_name:
        values = list(map(int, i.split("x")))
        values.sort()

        shortest_distance = 2*(int(values[0]) + int(values[1]))
        perfect_bow = int(values[0])*int(values[1])*int(values[2])

        foot += shortest_distance + perfect_bow

    return foot

def main():
    print(f"Part 1: {part1()}")
    print(f"Part 2: {part2()}")

if __name__ == "__main__":
    main()

File: 2015/j2/main.kt
package aoc

import java.nio.file.Files
import java.nio.file.Path
import java.nio.file.Paths

fun readInput(fileName: String): List<String> {
    val path: Path = Paths.get("").resolve(fileName)
    return Files.readAllLines(path)
}

fun part1(): Any {
    val inputName = readInput("input.txt")

    var foot = 0

    for (i in inputName) {
        val values = i.split("x").map { it.toInt() }.sorted()

        val additionnal = values[0].toInt()*values[1].toInt()
        val surfaces = 2*(values[0].toInt()*values[1].toInt() + values[0].toInt()*values[2].toInt() + values[1].toInt()*values[2].toInt())

        foot += additionnal + surfaces
    }

    return foot
}

fun part2(): Any {
    val inputName = readInput("input.txt")

    var foot = 0

    for (i in inputName) {
        val values = i.split("x").map { it.toInt() }.sorted()

        val shortest_distance = 2*(values[0].toInt() + values[1].toInt())
        val perfect_bow = values[0].toInt()*values[1].toInt()*values[2].toInt()

        foot += shortest_distance + perfect_bow
    }

    return foot
}

fun main() {
    println("Part 1: ${part1()}")
    println("Part 2: ${part2()}")
}

